### ISSUE TITLE
PM2 Daemon Dying in Prod

### ISSUE THREAD

**Issue by bartspedden** [Unitech/PM2#912]
Here's some environment information to start with

OS: Linux d6proddisp4 2.6.32-431.el6.x86_64 #1 SMP Sun Nov 10 22:19:54 EST 2013 x86_64 x86_64 x86_64 GNU/Linux
PM2: 0.12.1
NodeJS: v0.11.14

We are using the the generated init scripts to start/stop pm2.

We have 2 application running - each with 4 workers.

Over the past 3 weeks we have the PM2 God Daemon die 3 times with the following message showing up in the pm2.log

```
Error: channel closed
    at ChildProcess.target.send (child_process.js:413:26)
    at sendHelper (cluster.js:657:8)
    at RoundRobinHandle.handoff (cluster.js:198:3)
    at RoundRobinHandle.distribute (cluster.js:184:20)
```

Any ideas what might be causing this?

**Comment by Unitech** [comment]
Patch-Fixed on 0.12.4

**Comment by tuanna222** [comment]
_PM2 v2.1.5 died with these logs:_
```
2016-11-19 06:00:02: ===============================================================================
2016-11-19 06:00:02: --- PM2 global error caught ---------------------------------------------------
2016-11-19 06:00:02: Time                 : Sat Nov 19 2016 06:00:02 GMT+0700 (ICT)
2016-11-19 06:00:02: Cannot read property 'on' of undefined
2016-11-19 06:00:02: TypeError: Cannot read property 'on' of undefined
    at buildProcessTree (/usr/lib/node_modules/pm2/lib/TreeKill.js:81:14)
    at module.exports (/usr/lib/node_modules/pm2/lib/TreeKill.js:32:9)
    at Object.God.killProcess (/usr/lib/node_modules/pm2/lib/God/Methods.js:223:7)
    at Object.God.stopProcessId (/usr/lib/node_modules/pm2/lib/God/ActionMethods.js:261:9)
    at Object.God.deleteProcessId (/usr/lib/node_modules/pm2/lib/God/ActionMethods.js:313:9)
    at Worker.onListen (/usr/lib/node_modules/pm2/lib/God/Reload.js:134:18)
    at Worker.g (events.js:273:16)
    at emitOne (events.js:90:13)
    at Worker.emit (events.js:182:7)
    at listening (cluster.js:504:12)
2016-11-19 06:00:02: ===============================================================================
```
